src/simtools/visualization/simtel_event_plots.py

```diff
@@ -0,0 +1,612 @@
+        return None
+
+    w = np.asarray(waveforms)
+    if w.ndim == 3:
+        w = w[0]
+    _, n_samp = w.shape
+
+    peak_samples, pix_ids, found_count = _collect_peak_samples(
+        w, sum_threshold, peak_width, _signal
+    )
+    if peak_samples is None or pix_ids is None:
+        _logger.warning("No pixels exceeded sum_threshold for peak timing")
+        return None
+
+    mean_sample = float(np.mean(peak_samples))
+    std_sample = float(np.std(peak_samples))
+
+    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 4), dpi=300)
+
+    edges = _histogram_edges(n_samp, timing_bins)
+    et_name = getattr(getattr(event.trigger, "event_type", None), "name", "?")
+    tel = source.subarray.tel[tel_id]
+    tel_label = getattr(tel, "name", f"CT{tel_id}")
+    _draw_peak_hist(
+        ax1,
+        peak_samples,
+        edges,
+        mean_sample,
+        std_sample,
+        tel_label,
+        et_name,
+        pix_ids.size,
+        found_count,
+    )
+
+    readout = source.subarray.tel[tel_id].camera.readout
+    try:
+        dt = (1 / readout.sampling_rate).to(u.ns).value
+    except (AttributeError, ZeroDivisionError, TypeError):
+        dt = 1.0
+    t = np.arange(n_samp) * dt
+
+    ex_ids = pix_ids[: max(1, int(examples))]
+    for pid in ex_ids:
+        trace = w[int(pid)]
+        pks = _detect_peaks(trace, peak_width, _signal)
+        ax2.plot(t, trace, drawstyle="steps-mid", label=f"pix {int(pid)}")
+        if pks.size:
+            ax2.scatter(t[pks], trace[pks], s=10)
+    ax2.set_xlabel(TIME_NS_LABEL)
+    ax2.set_ylabel(R1_SAMPLES_LABEL)
+    ax2.legend(fontsize=7)
+
+    fig.tight_layout()
+
+    if return_stats:
+        stats = {
+            "considered": int(pix_ids.size),
+            "found": int(found_count),
+            "mean": float(mean_sample),
+            "std": float(std_sample),
+        }
+        return fig, stats
+    return fig
+
+
+def _prepare_waveforms_for_image(filename, tel_id, context_no_r1):
+    """Fetch R1 waveforms for one event/telescope and return prepared arrays.
+
+    Returns (w, n_pix, n_samp, source, event, tel_id) or None on failure.
+    """
+    # pylint:disable=import-outside-toplevel
+    import numpy as np
+    from ctapipe.io import EventSource
+
+    source = EventSource(filename, max_events=None)
+    event = _select_event_by_type(source)
+
+    r1_tel_ids = sorted(getattr(event.r1, "tel", {}).keys())
+    if r1_tel_ids:
+        tel_id = tel_id or r1_tel_ids[0]
+    else:
+        _logger.warning(f"Event has no R1 data for {context_no_r1}")
+        return None
+
+    waveforms = getattr(event.r1.tel.get(tel_id, None), "waveform", None)
+    if waveforms is None:
+        _logger.warning(NO_R1_WAVEFORMS_MSG)
+        return None
+
+    w = np.asarray(waveforms)
+    if w.ndim == 3:
+        w = w[0]
+    n_pix, n_samp = w.shape
+    return w, n_pix, n_samp, source, event, tel_id
+
+
+def plot_simtel_integrated_signal_image(
```

Comment:
What is the difference to plot_simtel_integrated_pedestal_image? This function seems to almost identical?